dask/distributed/tests/test_scheduler.py

```diff
@@ -2,32 +2,47 @@
 pytest.importorskip('zmq')
 pytest.importorskip('dill')
 
-from dask.distributed.scheduler import Scheduler
-from dask.distributed.worker import Worker
 import multiprocessing
-import itertools
+import pickle
+import re
 from datetime import datetime
 from contextlib import contextmanager
-from toolz import take
 from time import sleep
-import dill
-import pickle
-import re
 
 import zmq
+import dill
+
+from dask.distributed.scheduler import Scheduler
+from dask.distributed.worker import Worker
+from dask.compatibility import Queue
 
 context = zmq.Context()
 
 
 @contextmanager
-def scheduler():
-    s = Scheduler()
+def scheduler(kwargs={}):
+    s = Scheduler(**kwargs)
     try:
         yield s
     finally:
         s.close()
 
 
+@contextmanager
+def scheduler_and_workers(n=2, scheduler_kwargs={}, worker_kwargs={}):
```

Comment:
Added `scheduler_kwargs` and `worker_kwargs`, which is very verbose. But I don't have a better name.